File: src/main/java/my/study/repository/design/pattern/bridge/java/coffee/BaseCoffee.java
package my.study.repository.design.pattern.bridge.java.coffee;

import my.study.repository.design.pattern.bridge.java.additive.BaseAdditive;

/**
 * @author djh on  2018/9/25 14:10
 * @E-Mail 1544579459@qq.com
 */
abstract class BaseCoffee {
    BaseAdditive mBaseAdditive;

    BaseCoffee(BaseAdditive baseAdditive) {
        mBaseAdditive = baseAdditive;
    }

    abstract void showCoffeeInfo();
}

File: src/main/java/my/study/repository/design/pattern/bridge/kotlin/coffee/BaseCoffee.kt
package my.study.repository.design.pattern.bridge.kotlin.coffee

import my.study.repository.design.pattern.bridge.kotlin.additive.BaseAdditive


/**
 * @author djh on  2018/10/4 17:38
 * @E-Mail 1544579459@qq.com
 */
abstract class BaseCoffee(protected val baseAdditive: BaseAdditive) {

    abstract fun showCoffeeInfo()
}